draft-esdih-ccamp-layer0-types-ext-00.txt

```diff
@@ -0,0 +1,1344 @@
+   material or to cite them other than as "work in progress."
+
+   This Internet-Draft will expire on 26 August 2021.
+
+Copyright Notice
+
+   Copyright (c) 2021 IETF Trust and the persons identified as the
+   document authors.  All rights reserved.
+
+
+
+
+Beller, et al.           Expires 26 August 2021                 [Page 1]
+
+Internet-Draft         Layer 0 Types - Revision 2          February 2021
+
+
+   This document is subject to BCP 78 and the IETF Trust's Legal
+   Provisions Relating to IETF Documents (https://trustee.ietf.org/
+   license-info) in effect on the date of publication of this document.
+   Please review these documents carefully, as they describe your rights
+   and restrictions with respect to this document.  Code Components
+   extracted from this document must include Simplified BSD License text
+   as described in Section 4.e of the Trust Legal Provisions and are
+   provided without warranty as described in the Simplified BSD License.
+
+Table of Contents
+
+   1.  Introduction  . . . . . . . . . . . . . . . . . . . . . . . .   2
+     1.1.  Requirements Language . . . . . . . . . . . . . . . . . .   2
+     1.2.  Terminology . . . . . . . . . . . . . . . . . . . . . . .   3
+   2.  Extensions for the Layer 0 Types Module . . . . . . . . . . .   3
+   3.  Layer0 Types Revision 2 YANG CODE . . . . . . . . . . . . . .   4
+   4.  Acknowledgements  . . . . . . . . . . . . . . . . . . . . . .  22
+   5.  Contributors  . . . . . . . . . . . . . . . . . . . . . . . .  22
+   6.  IANA Considerations . . . . . . . . . . . . . . . . . . . . .  22
+   7.  Security Considerations . . . . . . . . . . . . . . . . . . .  22
+   8.  References  . . . . . . . . . . . . . . . . . . . . . . . . .  22
+     8.1.  Normative References  . . . . . . . . . . . . . . . . . .  22
+     8.2.  Informative References  . . . . . . . . . . . . . . . . .  23
+   Authors' Addresses  . . . . . . . . . . . . . . . . . . . . . . .  24
+
+1.  Introduction
+
+   YANG [RFC7950] is a data modeling language used to model
+   configuration data, state data, Remote Procedure Calls, and
+   notifications for network management protocols such as NETCONF
+   [RFC6241].  The YANG language supports a small set of built-in data
+   types and provides mechanisms to derive other types from the built-in
+   types.
+
+   This document introduces a collection of common data types derived
+   from the built-in YANG data types.  The derived types and groupings
+   are designed to be the common types applicable for modeling Traffic
+   Engineering (TE) features as well as non-TE features (e.g., physical
+   network configuration aspect) for Layer 0 optical networks in
+   model(s) defined outside of this document.
+
+1.1.  Requirements Language
+
+   The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+   "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+   document are to be interpreted as described in RFC 2119 [RFC2119].
```

Comment:
This draft is not using RFC2119 requirements language: this section can be removed